airbyte_cdk/sources/declarative/manifest_declarative_source.py

```
@@ -139,6 +144,35 @@ def __init__(
         # apply additional post-processing to the manifest
         self._post_process_manifest()
 
+        self._config: Mapping[str, Any]
+        self._spec_component: Optional[Spec] = None
+        spec = self._source_config.get("spec")
+        if spec:
+            if "type" not in spec:
+                spec["type"] = "Spec"
+            self._spec_component = self._constructor.create_component(SpecModel, spec, dict())
+            mutable_config = dict(config) if config else {}
+
+            if config_path:
+                self._spec_component.migrate_config(mutable_config)
+                try:
+                    if mutable_config != config:
+                        with open(config_path, "w") as f:
+                            json.dump(mutable_config, f)
+                        self.message_repository.emit_message(
+                            create_connector_config_control_message(mutable_config)
+                        )
+                        # We have no mechanism for consuming the queue, so we print the messages to stdout
+                        for message in self.message_repository.consume_queue():
+                            print(orjson.dumps(AirbyteMessageSerializer.dump(message)).decode())
+                except Exception as e:
```

I'm curious what types of errors we get here. I presume we can always have a chance of file not existing among others, but usually it's better to only catch the ones we expect and handle accordingly so that any unexpected errors are properly made loud

---

Yeah, I agree with you. Migrations *should* work or else fail loudly.